server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java

```diff
@@ -308,6 +308,21 @@ protected static int getMaxConcurrencyWriting(
   public StatusCode cacheShuffleData(
       String appId, int shuffleId, boolean isPreAllocated, ShufflePartitionedData spd) {
     refreshAppId(appId);
+    long partitionSize = getPartitionDataSize(appId, shuffleId, spd.getPartitionId());
+    long deltaSize = 0L;
+    for (ShufflePartitionedBlock spb : spd.getBlockList()) {
+      deltaSize += spb.getSize();
+    }
+    partitionSize += deltaSize;
+    if (shuffleBufferManager.isPartitionExceededMaxSize(partitionSize)) {
+      throw new InvalidRequestException(
```

Comment:
We need to return a new `StatusCode` instead of throwing an exception. When the client receive the new `StatusCode`, it should throw a `NotRetryException`.